c++/src/sargs/PredicateLeaf.cc

```diff
@@ -536,11 +536,17 @@ namespace orc {
             colStats.timestampstatistics().has_minimumutc() &&
             colStats.timestampstatistics().has_maximumutc()) {
           const auto& stats = colStats.timestampstatistics();
+          int64_t minimum = stats.minimumutc();
+          int64_t maximum = stats.maximumutc();
+          // To support predicates with ns precision we widen
+          // min and max ms range by 1 to ensure ns values are in range.
+          // e.g., for 1970-01-01 00:00:00.0010005 min
+          // becomes: 1970-01-01 00:00:00.000 and max: 1970-01-01 00:00:00.002
           result = evaluatePredicateRange(
             mOperator,
             literal2Timestamp(mLiterals),
-            stats.minimumutc(),
-            stats.maximumutc(),
+            minimum-1,
```

Comment:
Decreasing the minimum is probably needed, as the writers round towards negative infinity.